frontend/src/composables/useAmendingLaw.spec.ts

```diff
@@ -0,0 +1,88 @@
+import { defineStore } from "pinia"
+import { beforeEach, describe, expect, test, vi } from "vitest"
+import { nextTick, ref } from "vue"
+import { createTestingPinia } from "@pinia/testing"
+
+describe("useAmendingLaw", () => {
+  beforeEach(() => {
+    vi.resetModules()
+    vi.resetAllMocks()
+  })
+
+  test("should provide the amending law", async () => {
+    createTestingPinia({
+      stubActions: false,
+    })
+
+    vi.doMock("@/store/loadAmendingLawStore", () => ({
+      useAmendingLawsStore: vi.fn().mockReturnValue(
+        defineStore("loaded-amending-law", () => {
+          return {
+            loadedAmendingLaw: ref({
+              eli: "eli/bund/bgbl-1/2017/s419/2017-03-15/1/deu/regelungstext-1",
+              printAnnouncementGazette: "example",
+              digitalAnnouncementMedium: undefined,
+              publicationDate: "2023-01-01",
+              printAnnouncementPage: "1",
+              digitalAnnouncementEdition: undefined,
+              articles: [
+                {
+                  eli: "article eli 1",
+                  title: "article eli 1",
+                  enumeration: "1",
+                },
+              ],
+            }),
+            loadAmendingLawByEli: vi.fn(),
+          }
+        })(),
+      ),
+    }))
+
+    const { useAmendingLaw } = await import("./useAmendingLaw")
+
+    const eli = ref(
+      "eli/bund/bgbl-1/2017/s419/2017-03-15/1/deu/regelungstext-1",
+    )
+    const amendingLaw = useAmendingLaw(eli)
+
+    expect(amendingLaw.value?.eli).toBe(
+      "eli/bund/bgbl-1/2017/s419/2017-03-15/1/deu/regelungstext-1",
+    )
+    expect(amendingLaw.value?.publicationDate).toBe("2023-01-01")
+  })
+
+  test("should load the amending law when the eli changes", async () => {
+    const loadAmendingLawByEli = vi.fn()
+
+    createTestingPinia({
+      stubActions: false,
+    })
+
+    vi.doMock("@/store/loadAmendingLawStore", () => ({
+      useAmendingLawsStore: vi.fn().mockReturnValue(
+        defineStore("loaded-amending-law", () => {
+          return {
+            loadedAmendingLaw: ref(),
+            loadAmendingLawByEli,
+          }
+        })(),
+      ),
+    }))
+
+    const { useAmendingLaw } = await import("./useAmendingLaw")
```

Comment:
Question: why the async import?

Reply:
it allows each test case to mock the store independently. Otherwise the mocks are hoisted out of the individual test cases and are all applied to all test cases.